### PR TITLE
specgen: honor userns=auto from containers.conf

### PR DESCRIPTION
**Pull request by giuseppe**
when using the default userns value, make sure its value is parsed so
that userns=auto is parsed and the options for the storage are filled.

Closes: https://github.com/containers/podman/issues/12615

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: test/system/170-run-userns.bats
Comment (vrothberg) [inline test/system/170-run-userns.bats:60]: `$tmpdir` must be set.  Currently, we're creating a file at `/userns_auto.conf`.
```bash
tmpdir=$PODMAN_TMPDIR/userns-test
mkdir -p $tmpdir                
```

Comment (edsantiago) [reply test/system/170-run-userns.bats:60]: ...or just use `$PODMAN_TMPDIR` directly

Comment (giuseppe) [reply test/system/170-run-userns.bats]: thanks, adjusted to use `$PODMAN_TMPDIR`

File: test/system/170-run-userns.bats
Comment (vrothberg) [inline test/system/170-run-userns.bats:64]: Apologies for not spotting earlier. Could we instead of exporting set it as follows:
CONTAINERS_CONF=... run_podman

Otherwise, I am afraid of all subsequent calls using this containers.conf.

**Comment by rhatdan** [comment]
LGTM Once you make @vrothberg suggestion.

File: test/system/170-run-userns.bats
Comment (giuseppe) [reply test/system/170-run-userns.bats]: fixed.

I wrongly assumed each test would run in its own process

**Review by vrothberg — COMMENTED** [review]
/lgtm
/hold

**Comment by mheon** [comment]
LGTM

File: test/system/170-run-userns.bats
Comment (edsantiago) [inline test/system/170-run-userns.bats:64]: Could you add a comment explaining the significance of the magic constant?

And, could you add `--rm` just to be polite?

File: test/system/170-run-userns.bats
Comment (edsantiago) [inline test/system/170-run-userns.bats:57]: Is there a way you could compute this from the current context's `/proc/self/uid_map` and `/etc/subuid`?

**Review by edsantiago — CHANGES_REQUESTED** [review]
Finally getting to this. Thanks for fixing the `env` issue.

When I run this on my f35 laptop, as root, it fails:
```
   # .../podman run quay.io/libpod/testimage:20210610 grep -v 4294967295 /proc/self/uid_map
   time="2021-12-16T07:41:07-07:00" level=error msg="Cannot find mappings for user \"containers\": No subuid ranges found for user \"containers\" in /etc/subuid"
   Error: error creating container storage: could not find enough available IDs
   [ rc=125 (** EXPECTED 1 **) ]
```
If CI fails, then we're consistent, and the test needs fixing.

If CI _passes_, then the test in inconsistent, and we need to understand why.

File: test/system/170-run-userns.bats
Comment (giuseppe) [reply test/system/170-run-userns.bats]: we could parse `/etc/subuid`, dropped this line for an explicit check.

File: test/system/170-run-userns.bats
Comment (giuseppe) [reply test/system/170-run-userns.bats]: sure, added a comment.

Also added a commit to make the existing tests polite as well

File: test/system/170-run-userns.bats
Comment (edsantiago) [inline test/system/170-run-userns.bats:61]: Shouldn't this be `skip`?

File: test/system/170-run-userns.bats
Comment (edsantiago) [inline test/system/170-run-userns.bats:60]: Suggestion: add a colon, just to catch the possibility of `foo` vs `foo123`:
```diff
index 8b0591116..513f4ae41 100644
--- a/test/system/170-run-userns.bats
+++ b/test/system/170-run-userns.bats
@@ -57,13 +57,9 @@ function _require_crun() {
     skip_if_remote "userns=auto is set on the server"

     if is_rootless; then
-        if ! grep -q ^$(id -un) /etc/subuid; then
-           echo "there are no IDs allocated for the current user"
-        fi
+        egrep -q "^$(id -un):" /etc/subuid || skip "no IDs allocated for current user"
     else
-        if ! grep -q ^containers /etc/subuid; then
-           echo "there are no IDs allocated for the user 'containers'"
-        fi
+        egrep -q "^containers:" /etc/subuid || skip "no IDs allocated for user 'containers'"
     fi

     cat > $PODMAN_TMPDIR/userns_auto.conf <<EOF
```

File: test/system/170-run-userns.bats
Comment (edsantiago) [inline test/system/170-run-userns.bats:64]: Under what circumstances will `containers` exist in `/etc/subuid`? Because it doesn't on my laptop, and apparently (based on previous CI run) not in CI hosts either.

File: test/system/170-run-userns.bats
Comment (edsantiago) [inline test/system/170-run-userns.bats:75]: This fails on my laptop:
```
   $ .../podman run --rm quay.io/libpod/testimage:20210610 grep -v 4294967295 /proc/self/uid_map
            0          1       1024
   #/vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
   #| FAIL: exit code is 0; expected 1
   #\^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

**Review by edsantiago — CHANGES_REQUESTED** [review]
I still have serious concerns. None of this seems to work.

File: test/system/170-run-userns.bats
Comment (giuseppe) [reply test/system/170-run-userns.bats:64]: it must be added manually, but this is the user hardcoded in `containers/storage` to assign a range used by `auto` user namespaces

**Comment by giuseppe** [comment]
I think comments were addressed and tests are finally green

**Review by vrothberg — COMMENTED** [review]
/lgtm
/hold

@containers/podman-maintainers PTAL

**Pull request merged**
